cmd/dvotecli/commands/census.go

```diff
@@ -0,0 +1,210 @@
+	RunE:  addClaim,
+}
+
+var getRootCmd = &cobra.Command{
+	Use:   "root [id]",
+	Short: "return the root hash of the census merkle tree",
+	RunE:  getRoot,
+}
+
+var getSizeCmd = &cobra.Command{
+	Use:   "size [id]",
+	Short: "return the size of the census",
+	RunE:  getSize,
+}
+
+var genProofCmd = &cobra.Command{
+	Use:   "genproof [id] [key] [value]",
+	Short: "generate a proof of a given value",
+	RunE:  genProof,
+}
+
+func init() {
+	rootCmd.AddCommand(censusCmd)
+	censusCmd.AddCommand(censusAddCmd)
+	censusCmd.AddCommand(claimCmd)
+	censusCmd.AddCommand(getRootCmd)
+	censusCmd.AddCommand(getSizeCmd)
+	censusCmd.AddCommand(genProofCmd)
+	claimCmd.Flags().BoolVarP(&digested, "digested", "", false, "will digest value in the gateway if false")
+}
+
+func censusAdd(cmd *cobra.Command, args []string) error {
+
+	if err := checkSignKey(); err != nil {
+		return err
+	}
+
+	if len(args) < 1 {
+		return fmt.Errorf("you must provide a census id")
+	}
+
+	cl, err := client.New(host)
+	if err != nil {
+		return err
+	}
+	defer cl.Conn.Close(websocket.StatusNormalClosure, "")
+
+	var req types.MetaRequest
+	req.Method = "addCensus"
+	if len(args) > 1 {
+		req.PubKeys = args[1:]
+	}
+	req.CensusID = args[0]
+	resp, err := cl.Request(req, signKey)
+	if err != nil {
+		return err
+	}
+	if !resp.Ok {
+		return fmt.Errorf(resp.Message)
+	}
+	fmt.Printf("CensusID: %v\n", resp.CensusID)
+	fmt.Printf("URI: %v\n", resp.URI)
+
+	return nil
+}
+
+func addClaim(cmd *cobra.Command, args []string) error {
+
+	if err := checkSignKey(); err != nil {
+		return err
+	}
+
+	if len(args) < 2 {
+		return fmt.Errorf("you must provide a census id and a claim key")
+	}
+
+	cl, err := client.New(host)
+	if err != nil {
+		return err
+	}
+	defer cl.Conn.Close(websocket.StatusNormalClosure, "")
+
+	var req types.MetaRequest
+	req.Method = "addClaim"
+	req.CensusID = args[0]
+	req.Digested = digested
+	req.CensusKey = []byte(args[1])
+	resp, err := cl.Request(req, signKey)
+	if err != nil {
+		return err
+	}
+
+	if !resp.Ok {
+		return fmt.Errorf(resp.Message)
+	}
+	fmt.Printf("Root: %v\n", resp.Root)
+
+	return nil
+}
+
+func getRoot(cmd *cobra.Command, args []string) error {
+
+	if len(args) < 1 {
+		return fmt.Errorf("you must provide a census id")
+	}
+
+	cl, err := client.New(host)
+	if err != nil {
+		return err
+	}
+	defer cl.Conn.Close(websocket.StatusNormalClosure, "")
+
+	var req types.MetaRequest
+	req.Method = "getRoot"
+	req.CensusID = args[0]
+	resp, err := cl.Request(req, nil)
+	if err != nil {
+		return err
+	}
+
+	if !resp.Ok {
+		return fmt.Errorf(resp.Message)
+	}
+	fmt.Printf("Root: %v", hex.EncodeToString(resp.Root))
+	return nil
+}
+
+func getSize(cmd *cobra.Command, args []string) error {
+
+	if len(args) < 1 {
+		return fmt.Errorf("you must provide a census id")
+	}
+
+	cl, err := client.New(host)
+	if err != nil {
+		return err
+	}
+	defer cl.Conn.Close(websocket.StatusNormalClosure, "")
+
+	var req types.MetaRequest
```

Comment:
This is more idiomatic: `req := types.MetaRequest{ Method: "getSize", CensusID: args[0] }`